app/assets/javascripts/mpa/processors/peptide/ECPeptideProcessor.ts

```diff
@@ -0,0 +1,39 @@
+import {ProcessedPeptideContainer} from '../../ProcessedPeptideContainer';
+import {Count} from '../../counts/CountTable';
+import { ECCountTable } from '../../counts/ECCountTable';
+
+export namespace ECPeptideProcessor
+{
+    export function process(processedPeptides: ProcessedPeptideContainer): ECCountTable 
+    {
+        var peptideCounts = processedPeptides.countTable;
+        var pept2dataResponse = processedPeptides.response;
+
+        var ecCounts = new Map<string, Count>();
+        var ec2peptide = new Map<string, Set<string>>();
+        var peptide2ec = new Map<string, string[]>();
+
+        pept2dataResponse.response.forEach((data, peptide, _) => 
+        {
+            let fas = data.fa.data || [];
+            let peptideCount = peptideCounts.get(peptide)
+
+            Object.keys(fas)
+                .filter(term => term.startsWith("EC:"))
+                .forEach(term => 
+                    {
+                        ecCounts.set(term, (ecCounts.get(term) || 0) + peptideCount)
+                        
+                        if(!peptide2ec.has(peptide))
```

Comment:
Minor remark: I'm not a fan of if-statements without corresponding `{` and `}` brackets.